### ISSUE TITLE
Cause: duplicate entry: com/bumptech/glide/GeneratedAppGlideModule.class "

### ISSUE THREAD

**Issue by neeleshatale** [bumptech/glide#3527]
Hello,
when we tried to add new google places SDK, we encountered an eroor while generating build, 
**"Cause: duplicate entry: com/bumptech/glide/GeneratedAppGlideModule.class"**
Glide version:  **"3.7.0"**

PFA 
![glide_places_sdk](https://user-images.githubusercontent.com/20472096/52946594-5e34a780-339a-11e9-8f7e-8c550efca870.jpg)

**libraries used in my project:**
compile 'com.android.support:design:27.1.1'
    compile 'com.android.support:support-v4:27.1.1'
    compile 'com.android.support:appcompat-v7:27.1.1'
    compile 'com.android.support:multidex:1.0.1'
    compile 'com.github.nisrulz:recyclerviewhelper:25.0.0'
    compile 'com.code-troopers.betterpickers:library:2.5.1'
    compile 'com.jakewharton:butterknife:7.0.1'
    compile 'com.github.bumptech.glide:glide:3.7.0'
    compile 'com.mobsandgeeks:android-saripaar:2.0.2'
    compile 'com.github.castorflex.smoothprogressbar:library-circular:1.0.1'
    compile 'com.google.android.gms:play-services-maps:12.0.1'
    compile 'com.google.maps.android:android-maps-utils:0.5+'
    compile 'com.google.android.gms:play-services-location:12.0.1'
    compile 'com.amazonaws:aws-android-sdk-s3:2.6.14'
    compile 'joda-time:joda-time:2.9.3'
    compile 'com.edmodo:cropper:1.0.1'
    compile 'com.daimajia.easing:library:1.0.1@aar'
    compile 'com.daimajia.androidanimations:library:1.1.3@aar'
    compile 'org.greenrobot:greendao:3.0.1'
    compile 'com.lyft:scissors:1.1.1'
    compile 'com.google.code.gson:gson:2.8.2'
    compile 'com.squareup.retrofit2:retrofit:2.0.0-beta4'
    compile 'com.squareup.retrofit2:converter-gson:2.0.0-beta4'
    compile 'com.github.PhilJay:MPAndroidChart:v2.1.6'
    compile 'com.onesignal:OneSignal:[3.6.0, 3.99.99)'
    compile 'com.google.android.libraries.places:places-compat:1.0.0'
    compile 'com.android.support:cardview-v7:27.1.1'

**Comment by sjudd** [comment]
Are you sure you're not using a v4 version of Glide? 

The error implies that multiple libraries you're using have bundled in a generated AppGlideModule, which they should never do. Unfortunately it may not be straight forward to figure out which libraries. If you do find them, point them to: http://bumptech.github.io/glide/doc/configuration.html#avoid-appglidemodule-in-libraries

**Comment by im4eversmrt** [comment]
Im having the same issue here, any help ?

**Comment by geronimowheeli** [comment]
Me too having the same issue

**Comment by SumeetMoray** [comment]
me too having this issue !

**Comment by SumeetMoray** [comment]
Duplicate class com.bumptech.glide.GeneratedAppGlideModule found in modules classes.jar (com.github.bumptech.glide:glide:4.3.1) and jetified-glide-4.3.1.jar (com.github.bumptech.glide:glide:4.3.1)

**Comment by ManzhulaAnna** [comment]
having the same issue...

**Comment by JobGetabu** [comment]
Duplicate class com.bumptech.glide.GeneratedAppGlideModule found in modules classes.jar (com.github.bumptech.glide:glide:4.3.1) and jetified-glide-4.3.1.jar (com.github.bumptech.glide:glide:4.3.1)

**Comment by kanaksony** [comment]
Why is it closed, there are so many people complaining? I too am facing same issue?

**Comment by Endi327** [comment]
Having the same issue. Any solution?

**Comment by Endi327** [comment]
Put these two lines in your app level build.gradle:

`implementation 'com.github.bumptech.glide:glide:4.9.0'
annotationProcessor 'com.github.bumptech.glide:compiler:4.9.0'`

I don't really understand why is this solve the issue, I wasn't using glide maybe one of my libraries was.
I don't think this is the best solution, but if you'd to move on, this works and you can return to this issue later.

**Comment by JobGetabu** [comment]
Compiles but fails in run time when you open kayako

**Comment by sjudd** [comment]
So after the edit on the original issue, it looks like the original reporter was trying to use both Glide v3 and v4 in the same application, which won't work. Probably they were including v3 and one of their dependencies were including v4.

**Comment by chigup** [comment]
> Put these two lines in your app level build.gradle:
> 
> ```
> implementation 'com.github.bumptech.glide:glide:4.9.0'
> annotationProcessor 'com.github.bumptech.glide:compiler:4.9.0'
> ```
> 
> I don't really understand why is this solve the issue, I wasn't using glide maybe one of my libraries was.
> I don't think this is the best solution, but if you'd to move on, this works and you can return to this issue later.

It may be because your any other dependency is using glide

**Comment by SumeetMoray** [comment]
Hi everyone ... 

I was having this issue when I added google places library which includes this glide library !
I migrated my project to androidx (android jetpack)  and after than this issue was solved !

If you are using non-androidx project and using older support library ... please try migrating your project to androidx ... it might solve your problem !

-----------------------------------------------------------------------

Further, if it does not solve the problem try creating a new project in android studio with (include androidx) option and then copy all project files to new project ... this will surely solve the problem because this is what i did to solve my problem !

**Comment by mohammedelnady** [comment]
open it again

**Comment by jaelsonwagner** [comment]
I was facing the same issue. To solve it, I searched in my source code for classes with the @GlideModule annotation. The classes which extend from the AppGlideModule class. So, I kept only one of these classes with the annotation @GlideModule. The annotation processor searches for classes annotated with @GlideModule and then, try to generate the class GeneratedAppGlideModule. If they're more than one class with @GlideModule, we get that error.
